File: app/src/main/java/io/github/sithengineer/motoqueiro/network/RideWebService.java
package io.github.sithengineer.motoqueiro.network;

import io.github.sithengineer.motoqueiro.data.model.Ride;
import javax.inject.Inject;
import rx.Completable;
import rx.Scheduler;

public class RideWebService {

  private final NetworkService networkService;
  private final Scheduler ioScheduler;

  @Inject public RideWebService(NetworkService networkService, Scheduler ioScheduler) {
    this.networkService = networkService;
    this.ioScheduler = ioScheduler;
  }

  public Completable upload(Ride ride) {
    return networkService.upload(ride).subscribeOn(ioScheduler).toCompletable();
  }
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/network/RideWebService.kt
package io.github.sithengineer.motoqueiro.network

import io.github.sithengineer.motoqueiro.data.local.entity.Ride
import io.reactivex.Completable
import io.reactivex.Scheduler
import javax.inject.Inject

class RideWebService @Inject constructor(private val networkService: NetworkService,
    private val ioScheduler: Scheduler) {

  fun upload(ride: Ride): Completable {
    return networkService.upload(ride).subscribeOn(ioScheduler).ignoreElements()
  }
}
